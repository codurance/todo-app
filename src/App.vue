<template>
  <section id="todoapp">
    <header>
      <h1>TODO app</h1>
    </header>
    <section class="main">
      <add @todoAdded="updateList" />
      <list :todos="todos" />
    </section>
      <footer>
        <counter :todos="todos" />
      </footer>
  </section>
</template>

<script>
import Counter from '@/components/Counter.vue'
import List from '@/components/List.vue'
import Add from '@/components/Add.vue'

export default {
  name: 'App',
  components: {
    Counter,
    List,
    Add
  },
  data() {
    return {
      todos: []
    }
  },
  methods: {
    updateList(todo) {
      this.todos.push(todo)
    }
  }
}
</script>

<style>
body {
	font: 16px Helvetica, Arial, sans-serif;
	line-height: 1.4em;
	background: #f5f5f5;
	color: #4d4d4d;
	min-width: 230px;
	max-width: 550px;
	margin: 0 auto;
	font-weight: 300;
}
h1 {
	width: 100%;
	font-size: 100px;
	font-weight: 100;
	text-align: center;
	color: rgb(175, 47, 47);
}
</style>
