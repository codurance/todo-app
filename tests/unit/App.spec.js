import { mount } from '@vue/test-utils'
import App from '@/App.vue'

test('Adds a new TODO to the list and to the counter', async () => {
    const app = mount(App)

    const input = app.find('input')
    input.setValue('Breaking Bad')
    app.find('form').trigger('submit')
    await app.vm.$nextTick()

    expect(app.text()).toContain("Number of TODOs: 1")
    expect(app.find('li').text()).toBe('Breaking Bad')
})

test('Renders 2 items, the second one marked as done', async () => {
    function addItem(value) {
        input.setValue(value)
        app.find('form').trigger('submit')
    }

    const app = mount(App)
    const input = app.find('input')

    addItem('Breaking Bad')
    addItem('The Wire')
    //Update the view
    app.vm.$forceUpdate();
    await app.vm.$nextTick()
    //Mark the second item as done
    const secondItem = app.findAll('input[type="checkbox"]').at(1)
    secondItem.setChecked(true)
    await app.vm.$nextTick()

    expect(app.find('#todoapp')).toMatchSnapshot()
})
