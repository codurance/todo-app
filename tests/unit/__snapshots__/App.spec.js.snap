// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 2 items, the second one marked as done 1`] = `
<section id="todoapp">
  <header>
    <h1>TODO app</h1>
  </header>
  <section class="main">
    <form class="new-todo"><label for="new-todo">Add a todo</label> <input id="new-todo" placeholder="E.g. Walk the dog"> <button>Add</button></form>
    <ul>
      <li class="pending"><input type="checkbox">
        Breaking Bad
      </li>
      <li class="done"><input type="checkbox">
        The Wire
      </li>
    </ul>
  </section>
  <footer>
    <p>
      Number of TODOs: 2
    </p>
  </footer>
</section>
`;
